source/vibe/core/path.d

```diff
@@ -779,7 +772,30 @@ struct GenericPath(F) {
 
 		auto dst = appender!string;
 		Format.toString(newnodes, dst);
-		m_path = dst.data;
+
+		GenericPath ret;
+		ret.m_path = dst.data;
+		return ret;
+	}
+
+	///
+	unittest {
+		assert(PosixPath("foo/../bar").normalized == PosixPath("bar"));
+		assert(PosixPath("foo//./bar/../baz").normalized == PosixPath("foo/baz"));
+		assert(PosixPath("/foo/../bar").normalized == PosixPath("/bar"));
+		assert(WindowsPath(`\\PC/c$\foo/../bar/`).normalized == WindowsPath(`\\PC\c$\bar\`));
+		assert(PosixPath("/foo/bar").normalized == PosixPath("/foo/bar"));
+		assert(PosixPath("foo/bar/").normalized == PosixPath("foo/bar/"));
+	}
+
+
+	/** Replaces the path representation with its normalized form.
+
+		See `normalized` for a full description.
```

Comment:
```suggestion
		This is a simple wrapper around `normalized` which should be preferred. See `normalized` for a full description.
```

If you don't want to deprecate it.

Reply:
I've searched through the code base of a few projects and `normalize()` is being used in a surprising amount of cases, so I'd like to avoid breaking the API for this. I've adjusted the a bit wording here, but went with this solution.